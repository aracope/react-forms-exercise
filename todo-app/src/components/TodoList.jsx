import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

function TodoList() {
  const [todos, setTodos] = useState([]);

  const addTodo = (task) => {
    setTodos([...todos, { id: crypto.randomUUID(), task }]);
  };

  const removeTodo = (id) => {
    setTodos((todos) => todos.filter((todo) => todo.id !== id));
  };

  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      {todos.map((t) => (
        <Todo key={t.id} id={t.id} task={t.task} remove={removeTodo} />
      ))}
    </div>
  );
}

export default TodoList;

