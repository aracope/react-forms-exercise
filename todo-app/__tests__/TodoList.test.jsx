import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import TodoList from "../src/components/TodoList";

it("renders without crashing", () => {
    render(<TodoList />);
});

it("matches snapshot", () => {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("can add a new Todo", async () => {
    render(<TodoList />);

    await userEvent.type(screen.getByLabelText(/task/i), "Buy milk");
    await userEvent.click(screen.getByText("Add"));

    expect(screen.getByText("Buy milk")).toBeInTheDocument();
});

it("can remove a Todo", async () => {
    render(<TodoList />);

    await userEvent.type(screen.getByLabelText(/task/i), "Buy milk");
    await userEvent.click(screen.getByText("Add"));
    await userEvent.click(screen.getByText("X"));
    expect(screen.queryByText("Buy milk")).not.toBeInTheDocument();
});
