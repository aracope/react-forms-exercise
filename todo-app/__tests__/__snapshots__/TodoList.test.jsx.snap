// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="task"
      >
        Task:
      </label>
      <input
        aria-label="Task"
        id="task"
        name="task"
        placeholder="Enter a task"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
